### PR TITLE
Retry 3 times when joining a new network

### LINKED ISSUE #25: No Retry On Authentication After Entering Password.
After entering as password in the captive portal, it seems that there is only one try to associate. If there is a comms error like a 3=DEAUTH_LEAVING rather than a password error it still restarts the AP.  A retry would have connected.

The 'Trying to join known networks' path does have 3 retries and will connect in that particular case.
I have seen the NUC wifi take a bit of time to settle down when any of its modes have changed.

-Andy

### PR DESCRIPTION
**Pull request by pcarranzav**
Should fix #25

### CODE REVIEW

**Comment by lorenzo-stoakes** [comment]
LGTM, pending tests

File: src/app.coffee [hunk 1]
```diff
@@ -120,7 +120,16 @@ manageConnection = (retryCallback) ->
 			.then ->
 				dnsServer.kill()
 				console.log('Server closed and captive portal disabled')
-				wifi.joinAsync(req.body.ssid, req.body.passphrase)
+				Promise.new (resolve, reject) ->
+					async.retry {times: 3, interval: 1000}, (done) ->
+						wifi.joinAsync(req.body.ssid, req.body.passphrase)
+						.then ->
```
Comment (Page-) [inline src/app.coffee]: `.nodeify(done)`

File: src/app.coffee [hunk 2]
```diff
@@ -120,7 +120,16 @@ manageConnection = (retryCallback) ->
 			.then ->
 				dnsServer.kill()
 				console.log('Server closed and captive portal disabled')
-				wifi.joinAsync(req.body.ssid, req.body.passphrase)
+				Promise.new (resolve, reject) ->
```
Comment (Page-) [inline src/app.coffee]: `Promise.fromNode (callback) ->`

**Comment by pcarranzav** [comment]
Nice suggestions, @Page- - updated

**Comment by Page-** [comment]
LGTM

**Pull request merged**